packages/lit-node-client/src/lib/lit-node-client.ts

```diff
@@ -1027,6 +1031,87 @@ export class LitNodeClient {
     return rejectedPromises;
   };
 
+  /**
+   * Run lit action on a single deterministicly selected node. It's important that the nodes use the same deterministic selection algorithm.
+   *
+   * @param { JsonExecutionRequest } reqBody
+   *
+   * @returns { Promise<SuccessNodePromises | RejectedNodePromises> }
+   *
+   */
+  runOnSingleNode = async (
+    params: ExecuteJsProps
+  ): Promise<SuccessNodePromises> => {
+    const { code, authSig, jsParams, debug, sessionSigs, singleNode } = params;
+
+    // determine which node to run on
+    let ipfsId;
+    if (params.code) {
+      // hash the code to get IPFS id
+      const blockstore = new MemoryBlockstore();
+
+      let content;
+      if (typeof content === 'string') {
+        content = new TextEncoder().encode(content);
+      } else {
+        throwError({
+          message:
+            'Invalid code content type for single node execution.  Your code param must be a string',
+          error: LIT_ERROR.UNKNOWN_ERROR,
+        });
+      }
+
+      let lastCid;
+      for await (const { cid } of importer([{ content }], blockstore, {
+        onlyHash: true,
+      })) {
+        lastCid = cid;
+      }
+
+      ipfsId = lastCid;
+    } else {
+      ipfsId = params.ipfsId;
+    }
+
+    // should we mix in the jsParams?  to do this, we need a canonical way to serialize the jsParams object that will be identical in rust.
+    // const jsParams = params.jsParams || {};
+    // const jsParamsString = JSON.stringify(jsParams);
+
+    const hash = await sha256(`${ipfsId}`);
+    const hashAsNumber = BigNumber.from(hash);
+
+    // FIXME: we are using this.config.bootstrapUrls to pick the selected node, but we
+    // should be using something like the list of nodes from the staking contract
+    // because the staking nodes can change, and the rust code will use the same list
+    const nodeIndex = hashAsNumber
+      .mod(this.config.bootstrapUrls.length)
+      .toNumber();
+    const url = this.config.bootstrapUrls[nodeIndex];
+
+    log(`running on node ${nodeIndex} at ${url}`);
+
+    const reqBody: JsonExecutionRequest = this.getLitActionRequestBody(params);
+
+    // -- choose the right signature
+    let sigToPassToNode = this.getAuthSigOrSessionAuthSig({
+      authSig,
+      sessionSigs,
+      url,
+    });
+    reqBody.authSig = sigToPassToNode;
+
+    let executionResponse = await this.getJsExecutionShares(url, {
+      ...reqBody,
+    });
+
+    const successPromises: SuccessNodePromises = {
+      success: true,
+      values: [executionResponse as any],
```

Comment:
I added "as any" here to make this compile but i think it's wrong.  I think `values` should have type NodeShare[] but it wasn't clear to me how you transform the NodeCommandResponse into the NodeShare type inside `handleNodePromises`.